### ISSUE TITLE
Holding pieces causing rotation input drops

### ISSUE THREAD

**Issue by csq-curiosity** [o5k/tetrio-issues#180]
### Describe the bug/issue
If you hold a piece, there seems to be around a few frames of window where rotation inputs are dropped.

### Reproducing
It is advised to set ARE and line clear ARE to 20f to see the effect clearly.
1. Press hold.
2. Immediately after pressing hold, press rotate.
3. For a short amount of time after a piece was held, the rotation inputs are dropped, and the pieces are not rotated. (A coarse estimate of time seems to be around 6~8f.)

### Expected behavior
It is expected that the inputs are not dropped and the rotations are registered as soon as the hold piece spawns.

### Your setup
 - Windows 10
 - Chrome 80
 - The bug was tested with keyboard inputs. It may or may not be replicated with controller inputs.

### Additional context
- This bug also applies to initial hold system(IHS) and initial rotation system(IRS). While both IHS and IRS work fine independently, activating both IHS and IRS in conjunction, i.e. pressing hold and rotation keys at the same time, causes the held piece to spawn without rotation. The expected result is the held piece spawning with rotation.

**Comment by Cublex** [comment]
Probably a dumb comment, but on the IRS/IHS conflict I have a theory that IHS is processed after IRS, causing the piece to rotate _and then hold_, so the piece that you'd want to be rotated...doesn't.

**Comment by Wirtualnosc** [comment]
I tried to reproduce this on Opera 67.0.3575.97 (which is based on Chromium 80), but I couldn't reproduce that and I'm seeing expected behaviour... If that's any help. Tested with both pressing at the same time and in sequence.

**Comment by o5k** [comment]
> Probably a dumb comment, but on the IRS/IHS conflict I have a theory that IHS is processed after IRS, causing the piece to rotate _and then hold_, so the piece that you'd want to be rotated...doesn't.

This is true. However, I cannot reproduce the **first** issue (the dropped inputs), only that IHS will override IRS (the **second** issue). I'll fix the second definitely tho

**Comment by o5k** [comment]
The second issue has been fixed in the next patch. Since I cannot reproduce the first, I'm closing until more info arises.

**Comment by csq-curiosity** [comment]
So I tested again, with the knowledge that IHS has something to do with the bug. It turns out that when you press rotate while _holding down_ the hold key, the game will drop _all_ rotation inputs until the hold key is released. So it doesn't seem like IHS is the core issue but something with how the game handles hold inputs I guess?

I'll update this info to the original post accordingly.

**Comment by o5k** [comment]
Still can't reproduce :confused:

**Comment by csq-curiosity** [comment]
Tested, yet again. My keybinds for CCW, CW, and hold is DOT(.), SLASH(/) and RSHIFT respectively. Tried changing into COMMA(,), DOT(.) and SLASH(/), and the problem is resolved.

But it gets weirder. I tried SEMICOLON(;), APOSTROPHE(') and RSHIFT and the same problem arises. I then tried DOT(.), SLASH(/) and LSHIFT and the same problem is there. So the game seems to not understand combo inputs that involve shift properly. A similar issue exists for ctrl, but this seems to be more non-specific, as it recognizes certain combos properly, but fail to do with others.

In any case, it seems clear that there exists a completely different issue from this one, and need a new thread regarding this.
